src/edu/csus/ecs/pc2/core/scoring/DefaultPointScoringStandingsRecordComparator.java

```diff
@@ -0,0 +1,110 @@
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
+
+    private AccountNameComparator accountNameComparator = new AccountNameComparator();
+
+    private AccountList cachedAccountList;
+
+    /**
+     * Compares its two arguments for order. Returns a negative integer, zero, or a positive integer as the first argument is less
+     * than, equal to, or greater than the second.
+     * <p>
+     *
+     * The implementor must ensure that <tt>sgn(compare(x, y)) ==
+     * -sgn(compare(y, x))</tt> for all <tt>x</tt> and <tt>y</tt>.
+     * (This implies that <tt>compare(x, y)</tt> must throw an exception if and only if <tt>compare(y, x)</tt> throws an
+     * exception.)
+     * <p>
+     *
+     * The implementor must also ensure that the relation is transitive:
+     * <tt>((compare(x, y)&gt;0) &amp;&amp; (compare(y, z)&gt;0))</tt> implies <tt>compare(x, z)&gt;0</tt>.
+     * <p>
+     *
+     * Finally, the implementer must ensure that <tt>compare(x, y)==0</tt> implies that
+     * <tt>sgn(compare(x, z))==sgn(compare(y, z))</tt> for all <tt>z</tt>.
+     * <p>
+     *
+     * It is generally the case, but <i>not</i> strictly required that <tt>(compare(x, y)==0) == (x.equals(y))</tt>. Generally
+     * speaking, any comparator that violates this condition should clearly indicate this fact. The recommended language is "Note:
+     * this comparator imposes orderings that are inconsistent with equals."
+     *
+     * @return a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the
+     *         second.
+     * @throws ClassCastException
+     *             if the arguments' types prevent them from being compared by this Comparator.
+     */
+    @Override
+    public int compare(StandingsRecord o1, StandingsRecord o2) {
+        int status = 0;
+        long aLastSolvedTime, bLastSolvedTime;
+        double aScore, bScore;
+        int aClientHash, bClientHash;
+        String nameA, nameB;
+        int nameComparison;
+
+        StandingsRecord teamA = o1;
+        StandingsRecord teamB = o2;
+        aScore = teamA.getScore();
+        aLastSolvedTime = teamA.getLastSolved();
+        Account accountA = cachedAccountList.getAccount(teamA.getClientId());
+        nameA = accountA.getDisplayName();
+        aClientHash = teamA.getClientId().hashCode();
+        bScore = teamB.getScore();
+        bLastSolvedTime = teamB.getLastSolved();
+        Account accountB = cachedAccountList.getAccount(teamB.getClientId());
+        nameB = accountB.getDisplayName();
+        bClientHash = teamB.getClientId().hashCode();
+        nameComparison = accountNameComparator.compare(nameA.toLowerCase(), nameB.toLowerCase());
+
+        //
+        // Primary Sort = score (high to low)
+        // Secondary Sort = earliest submittal of last submission (low to high)
+        // Forth Sort = teamName (low to high)
+        // Fifth Sort = clientId (low to high)
```

Comment:
I don't understand why these two comments refer to "Forth [sic] Sort" and "Fifth Sort".  The two previous lines refer to "Primary [i.e. First] Sort" and "Secondary Sort".  So the next one should be "Third Sort" and the last one "Fourth Sort"... right?